csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/AssemblyInfo.cs

```diff
@@ -162,7 +176,39 @@ public static AssemblyInfo ReadFromFile(string filename)
                         publicKeyString.AppendFormat("{0:x2}", b);
 
                     var culture = def.Culture.IsNil ? "neutral" : reader.GetString(def.Culture);
-                    return new AssemblyInfo(filename, reader.GetString(def.Name), def.Version, culture, publicKeyString.ToString());
+                    Version? netCoreVersion = null;
+
+                    foreach (var attrHandle in def.GetCustomAttributes().Select(reader.GetCustomAttribute))
+                    {
+                        var ctorHandle = attrHandle.Constructor;
+                        if (ctorHandle.Kind != HandleKind.MemberReference)
+                        {
+                            continue;
+                        }
+
+                        var mHandle = reader.GetMemberReference((MemberReferenceHandle)ctorHandle).Parent;
+                        if (mHandle.Kind != HandleKind.TypeReference)
+                        {
+                            continue;
+                        }
+
+                        var name = reader.GetString(reader.GetTypeReference((TypeReferenceHandle)mHandle).Name);
+
+                        if (name is "TargetFrameworkAttribute")
+                        {
+                            var decoded = attrHandle.DecodeValue(new DummyAttributeDecoder());
```

Comment:
Should this `DecodeValue` call be wrapped in a try-catch? Can the `DummyAttributeDecoder` throw `NotImplementedException`s?

Reply:
I would actually like to be made aware if any of the `NotImplementedException`s are thrown.

Reply:
Yes, it would be good to know if this exception is thrown, but it seems a bit harsh to propgate the exception.
Maybe it is worth logging it as debug or information?